### PR TITLE
Add Changelog.md and Danger Github action to code review process

### LINKED ISSUE #3257: Add CHANGELOG.md and danger Github action
As mentioned in the developer meeting, I will update CHANGELOG.md and Danger check to main code base to remind contributors to update CHANGELOG every time the PR is submitted. We can use CHANGELOG for release notes in the future.

Danger (https://github.com/danger/danger) will also check CHANGELOG and give warnings to test, anti-pattern, etc.

### PR DESCRIPTION
**Pull request by pengyin-shan**
Fixes #3257 

### 1. Add Danger gem and Github action to the roadmap

Note that Danger is for reference. We can discuss changing the Danger configuration. Currently, the suggestion/reminder is:

1) PR prefers <= 1000 lines
2) If PR change >= 50, suggest adding a test
3) Remind to update CHANGELOG
4) `[WIP]` in the title will make this PR to be a draft pr
5) `#trivial` means no changelog is needed

An example:
Screenshot 2022-12-06 at 4 06 56 PM

### 2. CHANGELOG.md is ready for use

The workflow would be to add the description and link to the issue (if any) to the CHANGELOG every time a PR is submitted. I created the 'Changed', 'Added' and 'Fixed' sections for categorization purposes, but we are feeling to change to other formats. 
The person who's gonna do the release will add the release version as the header to the CHANGELOG. Later, developers will pull the CHANGELOG and continue to build above it.

Example:
Screenshot 2022-12-06 at 5 08 35 PM

By doing this, we will have a good record of the process, and whoever doing the release can use CHANGELOG as part of the release note. 

We can also consider adding this step to the contribution guide.

### CODE REVIEW

File: Dangerfile
Comment (briri) [inline Dangerfile:5]: should this be `/spec/` instead of `/test/`?

**Review by briri — APPROVED** [review]
looks good Wendy. I'm just wondering if the test check should use the RSpec directory instead of `/test/`

**Comment by briri** [comment]
and agree with @benjaminfaure that once this is ready we should merge it so we can start using the Changelog immediately

File: Dangerfile
Comment (pengyin-shan) [reply Dangerfile]: ops thank you for catching it! I'll update

**Comment by pengyin-shan** [comment]
Thank you @benjaminfaure @briri ! Will merge after all tests passed

**Pull request merged**
